src/components/filter/Filter.tsx

```diff
@@ -1,148 +1,155 @@
+    }
+  }, [filters]);
 
   if (!filterTypes?.length) {
     return null;
   }
 
   const isNoFilterSet =
-    filters.initial_start_date && !Object.keys(_omit(filters, Object.keys(initialFilters))).length;
+    filters.initial_start_date && !Object.keys(_omit(filters, Object.keys(queryVariables))).length;
 
   return (
     <>
-      <TouchableOpacity
-        onPress={() => setIsCollapsed(!isCollapsed)}
-        accessibilityLabel={`${isCollapsed ? texts.filter.showFilter : texts.filter.hideFilter} ${
-          a11yLabel.button
-        }`}
-        style={styles.button}
-      >
-        <BoldText small primary={!isCollapsed}>
-          {isCollapsed ? texts.filter.showFilter : texts.filter.hideFilter}
-        </BoldText>
-        <Icon.Filter
-          size={normalize(16)}
-          style={styles.icon}
-          color={!isCollapsed ? colors.primary : colors.darkText}
-        />
-      </TouchableOpacity>
-
-      <Collapsible collapsed={isCollapsed}>
-        <WrapperVertical style={styles.noPaddingTop}>
-          {filterTypes.map((item) => (
-            <WrapperVertical key={item.name} style={styles.noPaddingBottom}>
-              {item.type === FILTER_TYPES.DATE && (
-                <DateFilter
-                  filters={filters}
-                  setFilters={setFilters}
-                  containerStyle={{ width: device.width * 0.45 }}
-                  {...item}
-                  data={item.data as { name: keyof FilterProps; placeholder: string }[]}
-                />
-              )}
-
-              {item.type === FILTER_TYPES.DROPDOWN && item.data?.length && (
-                <DropdownFilter
-                  filters={filters}
-                  setFilters={setFilters}
-                  {...item}
-                  data={item.data as DropdownProps[]}
+      <Wrapper style={[withSearch && styles.noPadding]}>
+        <TouchableOpacity
+          onPress={() => setIsCollapsed(!isCollapsed)}
+          accessibilityLabel={`${isCollapsed ? texts.filter.showFilter : texts.filter.hideFilter} ${
+            a11yLabel.button
+          }`}
+          style={styles.button}
+        >
+          <BoldText small primary={!isCollapsed}>
+            {isCollapsed ? texts.filter.showFilter : texts.filter.hideFilter}
+          </BoldText>
+          <Icon.Filter
+            size={normalize(16)}
+            style={styles.icon}
+            color={!isCollapsed ? colors.primary : colors.darkText}
+          />
+        </TouchableOpacity>
+
+        {isOverlay ? (
+          <Modal
+            animationType="slide"
+            onRequestClose={() => setIsCollapsed(!isCollapsed)}
+            presentationStyle="pageSheet"
+            visible={!isCollapsed}
+          >
+            <Header
+              backgroundColor={colors.transparent}
+              centerComponent={{
+                text: texts.filter.filter,
+                style: { color: colors.darkText, fontSize: normalize(18), fontWeight: '700' }
```

Comment:
i think this needs to be more accurate with our normal header styles from `screenOptions`:

```
headerTitleStyle: {
    color: colors.lightestText,
    fontFamily: device.platform === 'ios' ? 'bold' : 'regular',
    fontSize: normalize(20),
    fontWeight: '400',
    lineHeight: normalize(29)
  },
```